lib/phoenix_token_auth/registration_flow.ex

```diff
@@ -0,0 +1,28 @@
+defmodule PhoenixTokenAuth.RegistrationFlow do
+  alias PhoenixTokenAuth.Mailer
+  alias PhoenixTokenAuth.Util
+
+  def user_created_by_username?(changeset) do
+    if changeset.valid? do
+      Util.repo.transaction(fn ->
+        case Util.repo.insert(changeset) do
+          {:ok, _} -> true
+          {:error, _} -> false
+        end
+      end)
+    end
+  end
+
+  def user_created_and_emailed?(changeset, confirmation_token, conn) do
+    if changeset.valid? do
+      Util.repo.transaction(fn ->
```

Comment:
i get that, but i just checked and Repo.insert won't hit the DB when the changeset is invalid anyway.
also, the phoenix generators produce controllers that don't check for valid but just match on the result of insert. i take them as templates when possible, because those people know a lot more than i do ;-)